tests/test_calc_week.py

```diff
@@ -0,0 +1,138 @@
+    assert calc_week(datetime.datetime(2015, 9, 8, tzinfo=pytz.utc)) == (2015, 'PRE', 4)
+    assert calc_week(datetime.datetime(2015, 9, 9, tzinfo=pytz.utc)) == (2015, 'REG', 1)
+    assert calc_week(datetime.datetime(2016, 9, 6, tzinfo=pytz.utc)) == (2016, 'PRE', 4)
+    assert calc_week(datetime.datetime(2016, 9, 7, tzinfo=pytz.utc)) == (2016, 'REG', 1)
+    assert calc_week(datetime.datetime(2017, 9, 5, tzinfo=pytz.utc)) == (2017, 'PRE', 4)
+    assert calc_week(datetime.datetime(2017, 9, 6, tzinfo=pytz.utc)) == (2017, 'REG', 1)
+    assert calc_week(datetime.datetime(2018, 9, 4, tzinfo=pytz.utc)) == (2018, 'PRE', 4)
+    assert calc_week(datetime.datetime(2018, 9, 5, tzinfo=pytz.utc)) == (2018, 'REG', 1)
+
+def test_reg_to_post_switches():
+    assert calc_week(datetime.datetime(2014, 12, 30, tzinfo=pytz.utc)) == (2014, 'REG', 17)
+    assert calc_week(datetime.datetime(2014, 12, 31, tzinfo=pytz.utc)) == (2014, 'POST', 1)
+    assert calc_week(datetime.datetime(2016, 1, 5, tzinfo=pytz.utc)) == (2015, 'REG', 17)
+    assert calc_week(datetime.datetime(2016, 1, 6, tzinfo=pytz.utc)) == (2015, 'POST', 1)
+    assert calc_week(datetime.datetime(2017, 1, 3, tzinfo=pytz.utc)) == (2016, 'REG', 17)
+    assert calc_week(datetime.datetime(2017, 1, 4, tzinfo=pytz.utc)) == (2016, 'POST', 1)
+    assert calc_week(datetime.datetime(2018, 1, 2, tzinfo=pytz.utc)) == (2017, 'REG', 17)
+    assert calc_week(datetime.datetime(2018, 1, 3, tzinfo=pytz.utc)) == (2017, 'POST', 1)
+    assert calc_week(datetime.datetime(2019, 1, 1, tzinfo=pytz.utc)) == (2018, 'REG', 17)
+    assert calc_week(datetime.datetime(2019, 1, 2, tzinfo=pytz.utc)) == (2018, 'POST', 1)
+
+def test__2018_schedule():
+    """
+    Just an extra set of tests to see that the coming season, as of this
+    writing, will work.
+    """
+    # POST4 to PRE0/HOF already tested
+
+    # PRE0/HOF to PRE1 already tested
+
+    assert calc_week(datetime.datetime(2018, 8, 14, tzinfo=pytz.utc)) == (2018, 'PRE', 1)
+    assert calc_week(datetime.datetime(2018, 8, 15, tzinfo=pytz.utc)) == (2018, 'PRE', 2)
+
+    assert calc_week(datetime.datetime(2018, 8, 21, tzinfo=pytz.utc)) == (2018, 'PRE', 2)
+    assert calc_week(datetime.datetime(2018, 8, 22, tzinfo=pytz.utc)) == (2018, 'PRE', 3)
+
+    assert calc_week(datetime.datetime(2018, 8, 28, tzinfo=pytz.utc)) == (2018, 'PRE', 3)
+    assert calc_week(datetime.datetime(2018, 8, 29, tzinfo=pytz.utc)) == (2018, 'PRE', 4)
+
+    # PRE4 to REG1 already tested
+
+    assert calc_week(datetime.datetime(2018, 9, 11, tzinfo=pytz.utc)) == (2018, 'REG', 1)
+    assert calc_week(datetime.datetime(2018, 9, 12, tzinfo=pytz.utc)) == (2018, 'REG', 2)
+
+    assert calc_week(datetime.datetime(2018, 9, 18, tzinfo=pytz.utc)) == (2018, 'REG', 2)
+    assert calc_week(datetime.datetime(2018, 9, 19, tzinfo=pytz.utc)) == (2018, 'REG', 3)
+
+    assert calc_week(datetime.datetime(2018, 9, 25, tzinfo=pytz.utc)) == (2018, 'REG', 3)
```

Comment:
Yes. Historically, NFL.com has switched weeks at 7AM UTC on wednesdays, so (most of the year) midnight Pacific. When I coded this, I decided that it was cleaner to just have a midnight switch on wednesdays instead of 7AM. UTC is used in several places throughout the application so I decided to use that here too.

IDK if it was a question just out of curiousity but I can easily change this if we feel that the switch should be pushed forward. I don't mind that, but I really think we should keep UTC for consistency and more importantly, the result should be the same no matter where in the world you run the application (disregarding unsynchronized system clocks 🙂)

Reply:
Really just out of curiosity...in some of my own scripts that I run to prepare for next week I was cheating/accounting for the week's change with a "if it's Tuesday, add 1 to the week returned from nflgame". Last night must have been the first time this season that I've done prep on Tuesday because my script was trying to tell me that I was preparing for week 5 (4 + 1).

I don't think `nflgame` needs to change. This was an artificial calculation I was doing on my part already...I just need to update my logic.

Reply:
Ah, cool :+1: A thought, maybe the nfl does all their fixing of wrong game data (#34) at the time where they at least used to switch weeks (wed 7AM UTC). If so, it would be useful to have this clearly defined in `nflgame` because it actually matters. As a bonus you won't have to change your code :wink:

Reply:
> maybe the nfl does all their fixing of wrong game data (#34) at the time where they at least used to switch weeks 

Nah. I've seen Thursday night games fixed by Friday morning and Sunday games fixed by Monday morning. The fixes, as far as I can tell, are not processed as a batch.